### PR TITLE
Add window.customElements

### PR DESCRIPTION
**Pull request by jpmedley**

### CODE REVIEW

File: api/Window.json [hunk 1]
```diff
@@ -47,6 +47,57 @@
...
+              "version_added": "54"
+            },
+            "chrome_android": {
+              "version_added": "54"
+            },
+            "edge": {
+              "version_added": false
+            },
+            "edge_mobile": {
+              "version_added": null
```
Comment (ExE-Boss) [inline api/Window.json]: Should be:
```json
"edge_mobile": {
  "version_added": false,
  "notes": "<a href='https://developer.microsoft.com/microsoft-edge/platform/status/customelements/'>Under consideration</a>"
}
```

File: api/Window.json [hunk 2]
```diff
@@ -47,6 +47,57 @@
...
+            },
+            "firefox": {
+              "version_added": false
```
Comment (ExE-Boss) [inline api/Window.json]: Should be (see #1294):
```json
"firefox": [
  {
    "version_added": "59",
    "flags": [
      {
        "type": "preference",
        "name": "dom.webcomponents.customelements.enabled",
        "value_to_set": "true"
      }
    ]
  },
  {
    "version_added": true,
    "version_removed": "59",
    "flags": [
      {
        "type": "preference",
        "name": "dom.webcomponents.enabled",
        "value_to_set": "true"
      },
      {
        "type": "preference",
        "name": "dom.webcomponents.customelements.enabled",
        "value_to_set": "true"
      }
    ]
  }
],
```

File: api/Window.json [hunk 3]
```diff
@@ -47,6 +47,57 @@
...
+            },
+            "firefox_android": {
+              "version_added": null
```
Comment (ExE-Boss) [inline api/Window.json]: Should be (see #1294):
```json
"firefox_android": [
  {
    "version_added": "59",
    "flags": [
      {
        "type": "preference",
        "name": "dom.webcomponents.customelements.enabled",
        "value_to_set": "true"
      }
    ]
  },
  {
    "version_added": true,
    "version_removed": "59",
    "flags": [
      {
        "type": "preference",
        "name": "dom.webcomponents.enabled",
        "value_to_set": "true"
      },
      {
        "type": "preference",
        "name": "dom.webcomponents.customelements.enabled",
        "value_to_set": "true"
      }
    ]
  }
],
```

File: api/Window.json [hunk 4]
```diff
@@ -47,6 +47,57 @@
...
+            },
+            "opera": {
+              "version_added": "41"
+            },
+            "opera_android": {
+              "version_added": "41"
+            },
+            "safari": {
+              "version_added": null
```
Comment (ExE-Boss) [inline api/Window.json]: Should be `"10.1"`

File: api/Window.json [hunk 5]
```diff
@@ -47,6 +47,57 @@
...
+            },
+            "safari_ios": {
+              "version_added": null
```
Comment (ExE-Boss) [inline api/Window.json]: ditto

File: api/Window.json [hunk 6]
```diff
@@ -47,6 +47,57 @@
...
+            },
+            "firefox": {
+              "version_added": false
+            },
+            "firefox_android": {
+              "version_added": null
+            },
+            "ie": {
+              "version_added": null
```
Comment (ExE-Boss) [inline api/Window.json]: Should be `false`

File: api/Window.json [hunk 7]
```diff
@@ -47,6 +47,57 @@
...
+              "version_added": "54"
+            },
+            "chrome": {
+              "version_added": "54"
+            },
+            "chrome_android": {
+              "version_added": "54"
+            },
+            "edge": {
+              "version_added": false
```
Comment (ExE-Boss) [inline api/Window.json]: Should be:
```json
"edge": {
  "version_added": false,
  "notes": "<a href='https://developer.microsoft.com/microsoft-edge/platform/status/customelements/'>Under consideration</a>"
}
```

File: api/Window.json [hunk 8]
```diff
@@ -47,6 +47,57 @@
           "deprecated": false
         }
       },
+      "customElements": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/Window/customElements",
+          "support": {
+            "webview_android": {
```
Comment (ExE-Boss) [inline api/Window.json]: This should be moved to the end of the file, as per #398.

**Comment by jpmedley** [comment]
PTAL

**Comment by ExE-Boss** [comment]
@chrisdavidmills As I explained in [my review](#discussion_r179315087), issue #398 is making it so that browsers must be sorted alphabetically in the `support` block.

**Comment by jpmedley** [comment]
I thought we were changing to alphabetical too. The only reason `webview_android` was at the top is because I asked that its name be changed from `android` to clarify what it is. Recent additions of Samsung, Node or whatever are alphabetized. It seems weird just to have one that's not.

**Pull request merged**